crates/language_models/src/provider/cloud.rs

```diff
@@ -88,6 +88,10 @@ pub struct AvailableModel {
     pub display_name: Option<String>,
     /// The size of the context window, indicating the maximum number of tokens the model can process.
     pub max_tokens: usize,
+    /// The number of thinking tokens to use for Gemini models.
+    /// If this is not set, the model decides on its own and the limit is 24576 tokens.
+    /// https://ai.google.dev/gemini-api/docs/thinking
+    pub thinking_budget: Option<usize>,
```

Comment:
This also means we can use the `mode` field below rather than introduce a new field just for google